python/ray/autoscaler/_private/resource_demand_scheduler.py

```diff
@@ -919,20 +919,21 @@ def get_bin_pack_residual(
         found = False
         node = None
         for i in range(len(nodes)):
+            if i in used:
+                continue
             node = nodes[i]
             if _fits(node, demand):
                 found = True
                 # In the strict_spread case, we can't reuse nodes.
                 if strict_spread:
-                    used.append(node)
-                    del nodes[i]
+                    used.add(i)
```

Comment:
Do not change the output order by using `del nodes[i]` and then concatenating `used` at the end. `_get_nodes_needed_for_request_resources` relies on the original order.